### ISSUE TITLE
`fs.walk()` should return relative paths, not absolute paths.

### ISSUE THREAD

**Issue by joehillen** [denoland/deno_std#3456]
**Describe the bug**

The current implementation of `walk()` with `followSymlinks = true` seems to
aggressively use `realPath()`. This is clearly the intended behavior based on
the [unit tests](https://github.com/denoland/deno_std/blob/main/fs/walk_test.ts). 
However, I think this is not the correct behavior. This is unusual behavior because it defeats
the purpose of symlinks. That is, links to directories and files should appear
to be children of the parent, even if their real location is not. I've never
seen a Unix application implicitly resolve and return a symlink'd absolute path.

This is exacerbated by the fact that `realPath()` throws an exception if the
the destination does not exist <https://github.com/denoland/deno_std/issues/3452> .

@Liamolucko Can you explain the reasoning for using `Deno.realPath()`?
<https://github.com/denoland/deno_std/commit/07be7159ce93b9f24a1f62c3496398d1d35b96e1#diff-820780ce193d0f4f9a9fd1f227979578c2d5607a43f40d3fb5c6eddd629f6af2R114>

**Steps to Reproduce**

An example use case for when the relative path would be desired would be:

```ts
import { walk } from "https://deno.land/std@0.192.0/fs/walk.ts";
import { join } from "https://deno.land/std@0.192.0/path/mod.ts";

Deno.mkdirSync("/tmp/test/x/y/z", { recursive: true });
Deno.createSync("/tmp/test/x/b.txt");
Deno.mkdirSync("/tmp/dest", { recursive: true });
Deno.createSync("/tmp/dest/a.txt");
Deno.symlinkSync("../../../../dest/a.txt", "/tmp/test/x/y/z/a.txt");

/**
 * Flatten a directory by copying all files to the root of the directory with
 * '/' replaced by '-'
 */
async function flat(src: string, dest: string) {
  await Deno.mkdir(dest, { recursive: true });

  for await (
    const entry of walk(src, {
      includeFiles: true,
      includeDirs: false,
      followSymlinks: true,
    })
  ) {
    let path = entry.path;
    if (path.startsWith(src)) {
      path = path.slice(src.length);
    }
    const fn = path
      .replace(/^\//, "") // remove leading `/`, if any
      .replace(/\//g, "-"); // replace all `/` with `-`
    await Deno.copyFile(entry.path, join(dest, fn));
  }
}

await flat("/tmp/test", "/tmp/flattened");
```

Result:

```txt
$ find /tmp/flattened
/tmp/flattened/
/tmp/flattened/tmp-dest-a.txt
/tmp/flattened/x-b.txt
```

**Expected behavior**

The resulting file is `/tmp/flattened/tmp-dest-a.txt` instead of `/tmp/flattened/x-y-z-a.txt`.

In the above example, I have no way of knowing that `/tmp/test/x/y/z/a.txt` exists
or where `tmp-dest-a.txt` came from because it was already resolved. I would
expect the `entry.path` to be relative to the parent directory and to be able
to call `Deno.realPath()` on my own if I wanted to resolve the symlink.

It would also be nice if `{ path: "/tmp/test/x/y/z/a.txt", isFile: true, isSymlink: true, ... }`
included `target: "../../../../dest/a.txt"`.

**Environment**

- OS: Arch Linux
- deno version: 1.34.1
- std version: 0.192.0

**Comment by Liamolucko** [comment]
> @Liamolucko Can you explain the reasoning for using `Deno.realPath()`?

It was 2 years ago so I don't really remember. There were already tests for that behaviour before my PR, so I wouldn't be surprised if I just added it when those tests failed.

**Comment by joehillen** [comment]
Resolved by #3679
